### DISCUSSION TITLE
Install Termux Packages on Android 6.0.1 Step By Step

### DISCUSSION THREAD

**Discussion by iTzVoko** [termux/termux-packages#11764] [Q&A] [answered]
Hello!

I'm trying to install Termux successfully on my **Android 6.0.1** device, and **I know** termux dropped the support from Android 5/6 devices.

*(I read all termux docs and search a lot)*

I've tried to update `sources.list` to `packages.termux.dev`  and setup everything,
but the main problem is when I try to install `nodejs-lts` packages, It installs nodejs v12 not v16, and the same for other packages.

So, how can I use grimler.se packages successfully on my Android 6.0.1? @Grimler91
Because I've tried that and I faced a problem with GPG public keys. 

I'm working on this version of Termux: [Termux-v0.79-offline-bootstraps](https://archive.org/download/termux-repositories-legacy)

Hope get the currect way to use Termux on my Android 6.0.1 device, because I like using termux rather than UserLAnd because it's lightweight and efficient.

More information:
```console
$ uname -a
Linux localhost 3.10.49-12221778 #1 SMP PREEMPT Tue Sep 5 19:54:15 KST 2017
armv7l Android

$ termux-info
Updatable packages:
All packages up to date
Subscribed repositories:
System information:
Linux localhost 3.10.49-12221778 #1 SMP PREEMPT Tue Sep 5 19:54:15 KST 2017 armv7l Android
Termux-packages arch:
arm
Android version:
6.0.1
Device manufacturer:
samsung
Device model:
SM-J500H
```

*I also have another device SM-J700 with armv8l*
Thanks.

**Comment by agnostic-apollo** [comment, marked as answer]
> search a lot)

https://github.com/termux/termux-app/discussions?discussions_q=android+6

And packages are compiled for android 7 in our main repos, and will not work on android 5/6. They will need to be compiled by yourself, since we don't plan on providing support for packages anymore, unless necessary for some specific package.

**Reply by iTzVoko** [reply]
Thanks, but that is not exactly what I need.

I want at least to download nodejs-lts v16.
but when I use `pkg install nodejs-lts` via [termux-app_v0.118.0-apt-android-5-github-debug_universal](https://github.com/termux/termux-app/discussions/2920#discussioncomment-3334717) it install legacy version (v12) but I want v16.

**Reply by agnostic-apollo** [reply]
Like I said, you will have to compile them yourself or look for someone who has already done so. Packages in our android 5/6 repo are 2 years old and are not updated and no plans either.

**Reply by iTzVoko** [reply]
If you're not busy, can you guide me on how to compile them?
because I need to use Termux

Currently, I've use [nodesource](https://github.com/nodesource/distributions), to install nodejs lts.
but even this doesn't work
```console
E: Package 'apt-transport-https' has no installation candidate
E: Unable to locate package lsb-release
```

**Reply by agnostic-apollo** [reply]
Instructions to build packages are available at https://github.com/termux/termux-packages/wiki/Building-packages

You will need to build with https://github.com/termux/termux-packages/tree/android-5 after porting patches from `master` branch. There are most likely going to be lot of issues, and not even sure if it can be done, and I don't really have time to help with that and nor I really use node. @thunder-coding the maintainer of `nodejs` would know better about potential issues. I think this discussion has been had before, search...

You could also try running `linux` distro binaries in `proot` instead.

**Reply by thunder-coding** [reply]
As the README suggests, it nodesource only supports Debian & Ubuntu based distributions. Aka it most probably does compile the node binaries for the kernel version contained in them, and doesn't consider Android (or Termux) support.

So yeah as Agnostic Apollo has suggested, try running it within a container like proot using proot-distro.

Talking about newer versions of node **natively** on Termux, they'll require latest versions of dependencies like libicu, libuv, and even libc+ (which currently is shipped from the NDK binaries), an NDK update will require rebuild of a lot of packages to be rebuilt. And even if you survive to do this, I would wish you good luck in trying to get rid of Android's Bionic's limitations and weirdness + some of the methods not available for old Android versions.

**Reply by iTzVoko** [reply]
Thanks for all!
I appreciate your time.
